Std/Data/ByteArray.lean

```diff
@@ -109,3 +110,34 @@ theorem get_extract_aux {a : ByteArray} {start stop} (h : i < (a.extract start s
 @[simp] theorem get_extract {a : ByteArray} {start stop} (h : i < (a.extract start stop).size) :
     (a.extract start stop)[i] = a[start+i]'(get_extract_aux h) := by
   simp [getElem_eq_data_getElem]
+
+/-! ### ofFn -/
+
+/--- `ofFn f` with `f : Fin n → UInt8` returns the byte array whose `i`th element is `f i`. --/
+def ofFn (f : Fin n → UInt8) : ByteArray where
+  data := .ofFn f
+
+@[simp] theorem ofFn_data (f : Fin n → UInt8) : (ofFn f).data = .ofFn f := rfl
+
+@[simp] theorem size_ofFn (f : Fin n → UInt8) : (ofFn f).size = n := by
+  simp [size]
+
+@[simp] theorem get_ofFn (f : Fin n → UInt8) (i : Fin (ofFn f).size) :
+    (ofFn f).get i = f (i.cast (size_ofFn f)) := by
+  simp [get, Fin.cast]
+
+@[simp] theorem getElem_ofFn (f : Fin n → UInt8) (i) (h : i < (ofFn f).size):
+    (ofFn f)[i] = f ⟨i, size_ofFn f ▸ h⟩ := get_ofFn ..
+
+private def ofFnAux (f : Fin n → UInt8) : ByteArray := go 0 (mkEmpty n) where
+  go (i : Nat) (acc : ByteArray) : ByteArray :=
+    if h : i < n then go (i+1) (acc.push (f ⟨i, h⟩)) else acc
+termination_by _ => n - i
+
+@[simp] theorem ofFnAux_data (f : Fin n → UInt8) :
+    (ofFnAux.go f i acc).data = Array.ofFn.go f i acc.data := by
+  rw [ofFnAux.go, Array.ofFn.go]; split; rw [ofFnAux_data, push_data]; rfl
+termination_by _ => n - i
+
+@[csimp] theorem ofFn_eq_ofFnAux : @ofFn = @ofFnAux := by
+  funext n f; ext; simp [ofFnAux, Array.ofFn, ofFnAux_data, mkEmpty]
```

Comment:
```suggestion
@[csimp] theorem ofFn_eq_ofFnAux : @ofFn = @ofFnAux := by
  let rec ofFnAux_data (f : Fin n → UInt8) :
      (ofFnAux.go f i acc).data = Array.ofFn.go f i acc.data := by
    rw [ofFnAux.go, Array.ofFn.go]; split; rw [ofFnAux_data, push_data]; rfl
  termination_by _ => n - i
  funext n f; ext; simp [ofFnAux, Array.ofFn, ofFnAux_data, mkEmpty]
```

Reply:
I should have paid attention to "or similar" :)

Reply:
The `@_ = @_` makes this change more tedious than necessary. If you have a trick to make it work, please let me know.

Reply:
Never mind, I got it!